src/test/regress/input/appendonly.source

```diff
@@ -811,3 +811,43 @@ insert into fix_ao_truncate_last_sequence select 1, 1 from generate_series(1, 5)
 select count(*) from fix_ao_truncate_last_sequence;
 abort;
 
+-- check reporting of missing attributes for AO table
+begin;
+create table test_reporting (id int) with (appendonly=true) distributed by (id);
+explain select xmin, * from test_reporting;
+abort;
+
+begin;
+create table test_reporting (id int) with (appendonly=true) distributed by (id);
+explain select xmax, * from test_reporting;
+abort;
+
+begin;
+create table test_reporting (id int) with (appendonly=true) distributed by (id);
+explain select cmin, * from test_reporting;
+abort;
+
+begin;
+create table test_reporting (id int) with (appendonly=true) distributed by (id);
+explain select cmax, * from test_reporting;
+abort;
+
+begin;
+create table test_reporting (id int) with (appendonly=true) distributed replicated;
+explain select xmin, * from test_reporting;
+abort;
+
+begin;
+create table test_reporting (id int) with (appendonly=true) distributed replicated;
+explain select xmax, * from test_reporting;
+abort;
+
+begin;
+create table test_reporting (id int) with (appendonly=true) distributed replicated;
+explain select cmin, * from test_reporting;
+abort;
+
+begin;
+create table test_reporting (id int) with (appendonly=true) distributed replicated;
+explain select cmax, * from test_reporting;
+abort;
```

Comment:
why do we need tests for replicated tables? it looks like the logic of working with them is the same as with other distributions.

Reply:
Good point. There was a different code path for replicated tables which had to be checked explicitly. Perhaps now it may be not so necessary but running more checks brings more confidence. I'd prefer to leave them.

Reply:
for logic related to system attributes in replicated tables have their own tests. I don't see the point in testing it here.

Reply:
> for logic related to system attributes in replicated tables have their own tests

where?

Reply:
> where?

https://github.com/arenadata/gpdb/blob/f6f95b6622a3fbdf441ec86164d5921b472694d7/src/test/regress/sql/rpt.sql#L337-L360

Reply:
Then why didn’t the error messages change from
```sql
ERROR:  column "xmin" does not exist
```
to these
```sql
ERROR: AO-table has no "xmin" column
```
in that test?

Reply:
> because they are not AO in this test

indeed!

> patch does not add/change logic specific to replicated tables

no! it changed logic for replicated AO-tables!

Reply:
okay. maybe then we should leave only one test for replicated tables to make sure that we get the same error as for other tables?

Reply:
What the point of removing tests?  Saving system resources and reducing execution time? It's milliseconds.

Reply:
the main reason is that this may confuse someone in the future, since looking at the list of changes in the code, it will be extremely unclear why we are testing replicated tables and explicitly opening transactions. if you insist that the tests of replicated tables should be left, then add a comment with motivation in the tests and in the description of the PR.

Reply:
> upd
> I still think that explicitly opening transactions does not make sense in this case.

Also, using transaction blocks is consistent with previous tests in this file.

Reply:
in those cases, the presence of a transaction is necessary to reproduce the problem.